New section on e-Service PDND (#528)
italia/eidas-it-wallet-docs: docs/en/e-service-pdnd.rst

.. include:: ../common/common_definitions.rst

e-Service PDND
+++++++++++++++++++

The `EIDAS-ARF`_ framework empowers Member States to establish the interfaces, terms, and conditions governing communication between Credential Issuers and Authentic Sources. In the Italian context, interoperability is established by leveraging the following guidelines:

    - "Linee Guida sull'interoperabilità tecnica delle Pubbliche Amministrazioni" (`MODI`_);
    - "Linee Guida sull'infrastruttura tecnologica della Piattaforma Digitale Nazionale Dati per l'interoperabilità dei sistemi informativi e delle basi di dati" (`PDND`_).

> To leverage the PDND, entities must formally subscribe, becoming **Participants** (*Aderenti*). Within the PDND infrastructure, Participants can assume the following roles:

Reviewer: ```suggestion
To leverage the PDND, entities MUST formally be subscribed becoming **Participants** (*Aderenti*). Within the PDND infrastructure, Participants MUST assume at least one of the the following roles:
```

========================================

This specification is based on the following set of requirements:

.. list-table::
    :widths: 10 70 20
    :header-rows: 1

    * - **ID**
      - **Description**
      - **Type**
    * - R1
      - The Consumer and the Provider have both subscribed to the PDND infrastructure.
      - Architectural
    * - R2
      - The communication between the Consumer and the Provider MUST ensure data integrity, authenticity, non-repudiation and replay protection.
      - Security
    * - R3
      - The Provider MAY require the Consumer to provide tracked data to complement the request. In that case, there MUST be a correlation between the tracked data and the Voucher.
      - Security
    * - R4
      - The e-Services MUST be implemented in REST, thus SOAP protocol MUST NOT be used.
      - Technical

`PDND`_ and `MODI`_ define several security patterns designed to enhance specific security properties in interactions between Participants. This specification adopts the following applicable security patterns:

.. list-table::
    :widths: 80 20
    :header-rows: 1

    * - **Security Pattern**
      - **Compliant With**

>     * - **[REST_JWS_2021_POP]** JWS POP Voucher Issuing Profile (*Annex 3 - Standards and technical details used for Voucher Authorization* [`PDND`_]): REQUIRED. It adds a proof of possession on the Voucher. The client using the Voucher to access an e-service MUST demonstrate the proof of possession of the private key whose public is attested on the Voucher.

Reviewer: ```suggestion
    * - **[REST_JWS_2021_POP]** JWS POP Voucher Issuing Profile (*Annex 3 - Standards and technical details used for Voucher Authorization* [`PDND`_]): REQUIRED. It adds a proof of possession on the Voucher. The Consumer using the Voucher to access an e-service MUST demonstrate the proof of possession of the private key whose public is attested on the Voucher.
```

      - R1, R2
    * - **[INTEGRITY_REST_02]** REST Payload Integrity in PDND (*Annex 2 - Security Patterns* [`MODI`_]): REQUIRED. It ensures the integrity of the payload of the REST Consumer request, within the PDND Infrastructure.
      - R2, R4
    * - **[AUDIT_REST_02]** Submission of audit data within the REST request with correlation (*Annex 2 - Security Patterns* [`MODI`_]): OPTIONAL. The Provider MAY request additional data tracked in the Consumer's domain, with a correlation between such data and the authentication method. In that case, this pattern MUST be used.
      - R3, R4

In addition, this specification defines and applies a custom security pattern:

.. list-table::
    :widths: 80 20
    :header-rows: 1

    * - **Security Pattern**
      - **Compliant With**
    * - REST Response Payload Integrity in PDND: REQUIRED. It ensures the integrity of the payload of the REST Provider response, within the PDND Infrastructure.
      - R2

> Some security patterns defined in `PDND`_ and `MODI`_ are not applicable as they do not comply with the requirements defined above:

Reviewer: ```suggestion
The following security patterns defined in `PDND`_ and `MODI`_ MUST NOT be used as they do not comply with the requirements defined above:
```

>     - The following patterns can only be used when the Consumer cannot subscribe to the PDND infrastructure (i.e., the trust between the Participants needs to be established in a direct form), thus not complying with **R1**:

Reviewer: ```suggestion
    - When the Consumer cannot subscribe to the PDND infrastructure (i.e., the trust between the Participants needs to be established in a direct form), thus not complying with **R1**, the following patterns MAY be used:
```

Author: Given the `MUST NOT` in the previous paragraph, this suggestion would change the meaning of the sentence. Moreover, for consistency with the following items (starting with "The following pattern(s) [...]"), we would keep the original version.

Key Request
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The Key Request is a ``GET`` HTTP request sent to the ``/keys/<kid>`` API. This request is used to retrieve a specific key identified by its unique ``kid``.

The Key Request MUST include the following HTTP header parameters:

.. list-table::
    :widths: 20 60 20
    :header-rows: 1

    * - **Parameter**
      - **Description**
      - **Reference**
    * - **Authorization**
      - Voucher released by the PDND Authorization Server.
      - [:rfc:`9449`]

Key Response
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The Interoperability API Endpoint responds with a ``200`` OK status code and a ``JWK`` [:rfc:`7517`] representing the public key with the provided ``kid``.

e-Service Usage
===================

Prerequisites
----------------------

The **Consumer** MUST comply with the following prerequisites:

    - Has obtained a valid Voucher to interact with the intended e-Service, related to a specific `Client e-service`.

The **Provider** MUST comply with the following prerequisites:

    - Has created a new key ring associated with the specific e-Service.
    - Has registered a key pair associated with the key ring.

.. note:: 

>     The key ring on the Provider's side is the reciprocal entity to the Client on the Consumer's side. This component serves as a store for cryptographic material, thus enabling Consumers to verify the integrity of responses transmitted by Providers.

Reviewer: ```suggestion
    The Provider's key ring is the counterpart to the Client on the Consumer's side. It stores cryptographic material, allowing Consumers to verify the integrity of responses from Providers.
```

I am not sure to have fully understood the sentence above, even if I try to simplify it
